pkg/recovery/etcd.go

```diff
@@ -153,28 +152,22 @@ func (s *etcdSelfhostedBackend) read(ctx context.Context) (*controlPlane, error)
 		return nil, err
 	}
 
-	d, err := s.getBytes(ctx, etcdTPRKey)
+	d, err := s.getBytes(ctx, etcdCRDKey)
 	if err != nil {
 		return nil, err
 	}
 
-	var tpr v1beta1.ThirdPartyResourceData
-	err = decode(s.decoder, d, &tpr)
+	var kubeetcd spec.EtcdCluster
+	err = json.Unmarshal(d, &kubeetcd)
```

Comment:
This is the only serious change and it seems to work. `migrate-etcd-cluster.json` end up with the `status` field and the json isn't pretty-printed, but I think that is the "expected" behavior.